app/components-react/windows/settings/Appearance.tsx

```diff
@@ -30,11 +31,11 @@ export function AppearanceSettings() {
 
   const { bind } = useModule(() => {
     function getSettings() {
-      return CustomizationService.state;
+      return CustomizationService.state.toObject() as CustomizationState;
     }
 
-    function setSettings(newSettings: typeof CustomizationService.state) {
-      CustomizationService.actions.setSettings(newSettings);
+    function setSettings(newSettings: CustomizationState) {
+      CustomizationService.actions.setSettings(newSettings as any);
```

Comment:
Ah yes this is one I was really struggling with.  This component is written in a fairly strange way where it wants to work with the entire settings object in a generic way.  Which in and of itself is not really a problem.  But something deep in the `useModule` typings is doing some weird transformation on it that makes it incompatible when passed back into `setSettings`, even though at runtime it's a completely valid object.  I struggled for a while figuring out what the issue was because it's a massive illegible typescript error.  I kind of gave up on this one.

Reply:
hopefully porting this over to zustand/FormFactory will be able to clear that up